### ISSUE TITLE
IPE Refactor Time Tracking While Coupled

### ISSUE THREAD

**Issue by akubaryk** [SWPC-IPE/WAM-IPE#312]
IPE_Wrapper uses the following calls:

```
    CALL ipe % time_tracker % Set_Date( year, month, day )
    CALL ipe % time_tracker % Set_HourMinute( hour, minute )
```

The `time_tracker % Set_HourMinute` subroutine further calls `time_tracker % Calculate_UTime( )`, and that sets `utime = REAL( time_tracker % hour*3600 + time_tracker % minute*60 )`. As an (extreme) example, consider we are starting from 23UTC and integrating for two hours; in the namelist, start_time = 82800, in the model utime = 82800 to begin, and end_time will be 82800 + 3600*2 = 90000. But IPE_Wrapper will overwrite utime = 0 at 00UTC, and end_time will never be reached.

I haven't come up with a generalized solution yet, but it's notable that while running with IPE_Driver instead of IPE_Wrapper (standalone vs coupled, as far as I can tell), this issue doesn't occur... none of the IPE_Driver time_tracker calls reset utime in the same way.

**Comment by akubaryk** [comment]
The time_tracker calls in IPE_Wrapper's Update_IPE are entirely superfluous and could be removed, likely solving this issue outright, but @rmontuoro has pointed out that we should use the clock provided by the NUOPC layer when running coupled to ensure consistency.

**Comment by rmontuoro** [comment]
Variable `utime` might be used in conflicting ways throughout IPE. While, for instance, MSIS requires UTC time for consistency with the local apparent solar time (`STL`), other parts of the code may use `utime` as the total elapsed time (s) since the model started.

Commit e84727a introduces a new `elapsed_sec` variable in IPE `Time` class to help keep track of the model's elapsed time.

We should clarify the role of `utime` and ensure the `time_tracker` can accurately provide all required information.

Also, the relationship between the forcing `start_time` and the `Time` class needs to be further defined.

**Comment by akubaryk** [comment]
Latest commit transitions elapsed_sec into use for the forcing class, and also subtracts 86400 from utime in the case that the hour > 24, so utime is tracking the elapsed seconds in the day.

Changes certainly need to be tested, but the advancement of the forcing parameters now works again, and I think the treatment of utime should be consistent where the pieces of the model are expecting it.

**Comment by akubaryk** [comment]
Recent commits have addressed time-handling in restart cases (7eb035a and 56fcaf3), MSIS (0e5769a), and the dynamo solver (a4c6875). This appears to be resolved now.
